src/src/pages/Metrics/components/ChartPopover/ChartPopoverAdvanced.tsx

```diff
@@ -0,0 +1,278 @@
+import React from 'react';
+
+import { Checkbox, TextField } from '@material-ui/core';
+import {
+  CheckBox as CheckBoxIcon,
+  CheckBoxOutlineBlank,
+} from '@material-ui/icons';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { Button, Box } from '@material-ui/core';
+
+import { Text } from 'components/kit';
+import ErrorBoundary from 'components/ErrorBoundary';
+
+import {
+  IGroupingCondition,
+  IGroupingSelectOption,
+} from 'types/services/models/metrics/metricsAppModel';
+import { IGroupingPopoverAdvancedProps } from 'types/components/GroupingPopover/GroupingPopover';
+
+import './ChartPopoverAdvanced.scss';
+
+export enum IOperator {
+  '==' = '==',
+  '!=' = '!=',
+  '>' = '>',
+  '<' = '<',
+  '>=' = '>=',
+  '<=' = '<=',
+}
+
+function ChartPopoverAdvanced({
+  onGroupingConditionsChange,
+  groupingData,
+  groupingSelectOptions,
+}: IGroupingPopoverAdvancedProps): React.FunctionComponentElement<React.ReactNode> {
+  // Current value for selectedField
+  const [inputValue, setInputValue] = React.useState('');
+
+  // Controls for the condition inputs
+  const [selectedField, setSelectedField] =
+    React.useState<IGroupingSelectOption | null>(null);
+
+  const [selectedOperator, setSelectedOperator] =
+    React.useState<IOperator | null>(IOperator['==']);
+
+  const [selectedValue, setSelectedValue] = React.useState<string>('');
+
+  // Array of grouping conditions
+  const [conditions, setConditions] = React.useState<IGroupingCondition[]>(
+    groupingData?.conditions || [],
+  );
+
+  const operators = [
+    IOperator['=='],
+    IOperator['!='],
+    IOperator['>'],
+    IOperator['<'],
+    IOperator['>='],
+    IOperator['<='],
+  ];
+
+  function onAddCondition() {
+    // Create a new condition object
+    const condition: IGroupingCondition = {
+      fieldName: selectedField?.label || '',
+      operator: selectedOperator ?? IOperator['=='],
+      value: selectedValue,
+    };
+
+    // If condition fieldName is not in array, add it, else replace
+    const conditionIndex = conditions.findIndex(
```

Comment:
maybe slightly more efficient to use dictionary/map keyed by the fieldname -- then you don't have to find it or have two branches to handle find result.

Reply:
Switching for a dictionary would slow down the rendering process (as that would require converting the dictionary into an array in every render in order to get the individual keys for the deletion function). Since rendering happens more often than changing the conditions, I think the array might be the better choice